BardMusicPlayer.Seer/Lib/Machina/TCPNetworkMonitor.cs

```diff
@@ -155,30 +139,26 @@ private void ProcessDataLoop(CancellationToken token)
 
     private void ProcessNetworkData()
     {
-        byte[] tcpbuffer;
-        byte[] payloadBuffer;
-
-        for (int i = 0; i < _connectionManager.Connections.Count; i++)
+        foreach (var connection in _connectionManager.Connections)
         {
-            TCPConnection connection = _connectionManager.Connections[i];
+            using var socket = connection.Socket;
             CapturedData data;
-
-            while ((data = connection.Socket.Receive()).Size > 0)
+            while ((data = socket.Receive()).Size > 0)
             {
                 connection.IPDecoderSend.FilterAndStoreData(data.Buffer, data.Size);
-
+                byte[] tcpbuffer;
```

Comment:
what is the point of allocating this here when it is done so by the while loop?

Reply:
Rider suggested the declaration can be moved closer to usage, which moves it right above the two while loop checks inside the main loop. The inside loops use the tcpbuffer.